syft/pkg/cataloger/golang/parse_go_bin.go

```diff
@@ -0,0 +1,125 @@
+package golang
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/anchore/syft/syft/pkg"
+)
+
+const packageIdentifier = "dep"
+
+func parseGoBin(path string, reader io.ReadCloser) ([]pkg.Package, error) {
+	pkgsSlice := make([]pkg.Package, 0)
+
+	// Identify if bin was compiled by go
+	x, err := openExe(reader)
+	if err != nil {
+		return pkgsSlice, err
+	}
+
+	_, mod := findVers(x)
+	fields := strings.Fields(mod)
+
+	// slice off root pacakge info
+	var separator int
+	for x, field := range fields {
+		if field == packageIdentifier {
+			separator = x - 1
+			break
+		}
+	}
+
+	fields = fields[separator:]
+
+	// filter deps: [dep, name, version, sha]
+	for x, field := range fields {
+		if field == packageIdentifier {
+			pkgsSlice = append(pkgsSlice, pkg.Package{
+				Name:     fields[x+1],
+				Version:  fields[x+2],
+				Language: pkg.Go,
+				Type:     pkg.GoModulePkg,
+			})
+		}
+	}
+
+	sort.SliceStable(pkgsSlice, func(i, j int) bool {
+		return pkgsSlice[i].Name < pkgsSlice[j].Name
+	})
+
+	return pkgsSlice, nil
+}
+
+// The build info blob left by the linker is identified by
```

Comment:
regarding the files we copied in from the internal go code, a couple of asks:
- can we keep them in their own (original) files? In this way if we need to incorporate upstream updates it makes it easier
- at the top of each file copied into our repo, can you add a note that mentions something to the affect of "this code was copied from <GH link to file>"?